examples/wifinina/connect/main.go

```diff
@@ -0,0 +1,134 @@
+// This example connects to Access Point and prints some info
+package main
+
+import (
+	"encoding/binary"
+	"fmt"
+	"machine"
+	"time"
+
+	"tinygo.org/x/drivers/wifinina"
+)
+
+// access point info
+const ssid = ""
+const pass = ""
+
+// these are the default pins for the Arduino Nano33 IoT.
+// change these to connect to a different UART or pins for the ESP8266/ESP32
+var (
+
+	// these are the default pins for the Arduino Nano33 IoT.
+	uart = machine.UART2
+	tx   = machine.NINA_TX
+	rx   = machine.NINA_RX
+	spi  = machine.NINA_SPI
+
+	// this is the ESP chip that has the WIFININA firmware flashed on it
+	adaptor *wifinina.Device
+)
+
+func setup() {
+	uart.Configure(machine.UARTConfig{TX: tx, RX: rx})
+
+	// Configure SPI for 8Mhz, Mode 0, MSB First
+	spi.Configure(machine.SPIConfig{
+		Frequency: 8 * 1e6,
+		SDO:       machine.NINA_SDO,
+		SDI:       machine.NINA_SDI,
+		SCK:       machine.NINA_SCK,
+	})
+
+	adaptor = wifinina.New(spi,
+		machine.NINA_CS,
+		machine.NINA_ACK,
+		machine.NINA_GPIO0,
+		machine.NINA_RESETN)
+	adaptor.Configure()
+}
+
+func main() {
+
+	setup()
+
+	waitSerial()
+
+	connectToAP()
+
+	for {
+		println("---------------------------------")
+		printIPs()
+		printTime()
+		printMacAddress()
+		time.Sleep(10 * time.Second)
+	}
+
+}
+
+func printIPs() {
+	ip, subnet, gateway, err := adaptor.GetIP()
+	if err != nil {
+		println("IP: Unknown (error: ", err.Error(), ")")
+		return
+	}
+	println("IP: ", ip.String())
+	println("Subnet: ", subnet.String())
+	println("Gateway IP: ", gateway.String())
+}
+
+func printTime() {
+	print("Time: ")
+	t, err := adaptor.GetTime()
+	if err != nil {
+		println("Unknown (error: ", err.Error(), ")")
+	}
+	fmt.Printf("%v\n", time.Unix(int64(t), 0))
```

Comment:
Tiny nitpick, you need `\r\n` to get the formatting nice.

Reply:
Sure, will fix later today